From f29058075bad5eb50b7ffdb4b1e19984b9ae4316 Mon Sep 17 00:00:00 2001
From: Yannick Vaucher <yannick.vaucher@camptocamp.com>
Date: Tue, 22 Oct 2019 17:48:54 +0200
Subject: [PATCH] Fix base amount for global_rounding on invoice tax

Additional precision must be done only on tax computation
Base amount must be set to currency precision in order to have
a sum of tax lines base amount equals to invoice total untaxed_amount.
---
 addons/account/models/account.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/addons/account/models/account.py b/addons/account/models/account.py
index 1e739007490..b0944bac8e0 100644
--- a/addons/account/models/account.py
+++ b/addons/account/models/account.py
@@ -775,15 +775,18 @@ class AccountTax(models.Model):
             round_tax = bool(self.env.context['round'])
             round_total = bool(self.env.context['round'])
 
-        if not round_tax:
-            prec += 5
-
         base_values = self.env.context.get('base_values')
         if not base_values:
             total_excluded = total_included = base = round(price_unit * quantity, prec)
         else:
             total_excluded, total_included, base = base_values
 
+        # Additional precision must be done only on tax computation
+        # Base amount must be set to currency precision in order to have
+        # a sum of tax lines base amount equals to invoice untaxed_amount.
+        if not round_tax:
+            prec += 5
+
         # Sorting key is mandatory in this case. When no key is provided, sorted() will perform a
         # search. However, the search method is overridden in account.tax in order to add a domain
         # depending on the context. This domain might filter out some taxes from self, e.g. in the
-- 
2.17.1

